discovery/hetzner/hcloud_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hetzner
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/util/testutil"
+	"testing"
+)
+
+type hcloudSDTestSuite struct {
+	Mock *SDMock
+}
+
+func (s *hcloudSDTestSuite) SetupTest(t *testing.T) {
+	s.Mock = NewSDMock(t)
+	s.Mock.Setup()
+
+	s.Mock.HandleHcloudServers()
+	s.Mock.HandleHcloudNetworks()
+}
+
+func TestHCloudSDRefresh(t *testing.T) {
+	suite := &hcloudSDTestSuite{}
+	suite.SetupTest(t)
+
+	cfg := DefaultSDConfig
+	cfg.HTTPClientConfig.BearerToken = hcloudTestToken
+	cfg.hcloudEndpoint = suite.Mock.Endpoint()
+
+	d, err := newHcloudDiscovery(&cfg, log.NewNopLogger())
+	testutil.Ok(t, err)
+
+	targetGroups, err := d.refresh(context.Background())
+	testutil.Ok(t, err)
+	testutil.Equals(t, 1, len(targetGroups))
+
+	targetGroup := targetGroups[0]
+	testutil.Assert(t, targetGroup != nil, "targetGroup should not be nil")
+	testutil.Assert(t, targetGroup.Targets != nil, "targetGroup.targets should not be nil")
+	testutil.Equals(t, 2, len(targetGroup.Targets))
+
+	for i, labelSet := range []model.LabelSet{
+		{
+			"__address__":                                            model.LabelValue("1.2.3.4:80"),
+			"__meta_hetzner_platform":                                model.LabelValue("hcloud"),
+			"__meta_hetzner_server_id":                               model.LabelValue("42"),
+			"__meta_hetzner_server_name":                             model.LabelValue("my-server"),
+			"__meta_hetzner_server_status":                           model.LabelValue("running"),
+			"__meta_hetzner_ipv4":                                    model.LabelValue("1.2.3.4"),
+			"__meta_hetzner_ipv6":                                    model.LabelValue("2001:db8::/64"),
+			"__meta_hetzner_datacenter":                              model.LabelValue("fsn1-dc8"),
+			"__meta_hetzner_hcloud_image_name":                       model.LabelValue("ubuntu-20.04"),
+			"__meta_hetzner_hcloud_image_description":                model.LabelValue("Ubuntu 20.04 Standard 64 bit"),
+			"__meta_hetzner_hcloud_datacenter_location":              model.LabelValue("fsn1"),
```

Comment:
```suggestion
			"__meta_hetzner_datacenter_location":              model.LabelValue("fsn1"),
```

to match __meta_hetzner_datacenter

Reply:
The Location is only returned from the Cloud API. The Robot API doesn't know about "locations" at all. So I moved it to a cloud-specific label.

Reply:
ack